core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -159,49 +159,34 @@ class FileMessageSet private[kafka](@volatile var file: File,
   /**
    * Search forward for the message whose timestamp is greater than or equals to the target timestamp.
    *
-   * The search will stop immediately when it sees a message in format version before 0.10.0. This is to avoid
-   * scanning the entire log when all the messages are still in old format.
```

Comment:
It seems like we were previously worried about scanning the entire log. How come that is no longer the case?